students/Briadean/lesson06/Mailroom4_test.py

```diff
@@ -0,0 +1,53 @@
+#!/usr#!/usr/bin/env python3
+
+import os
+from contextlib import redirect_stdout
+import Mailroom4 as Mailroom
+import io
+
+
+def test_accept_donation():
+    Mailroom.accept_donation(name="Bri")
+    assert "Bri" in Mailroom.donors_dict.keys()
+
+
+def test_list_donors():
+    file = io.StringIO()
+    with redirect_stdout(file):
+        print(Mailroom.list_donors())
+    contents = file.getvalue()
+    try:
+        assert ("Retsuko Rarecho" and
+                "Fenneko Inoue" and
+                "Haida Kato" and
+                "Gori Tsuruta" and
+                "Washimi Koiwasaki") in contents
+    finally:
+        file.close()
+
+
+def test_generate_report():
+    file = io.StringIO()
+    with redirect_stdout(file):
+        print(Mailroom.generate_report())
+    contents = file.getvalue()
+    try:
+        for name, donation in Mailroom.donors_dict.items():
+            assert name in contents
+            assert str(sum(donation)) in contents
+            assert str(len(donation)) in contents
+    finally:
+        file.close()
+
+
+def test_write_letters_disk():
+    files = Mailroom.write_letters_disk()
+    disk_files = os.listdir()
+    # Complaining path not specified, throwing TypeError 'NoneType' not iterable for line 47.I thought this defaults CWD
+    assert [file in disk_files for file in files]
```

Comment:
It looks like files is a command to write the letters, so this list comp is trying to iterate through that, but can't cause files is a Nonetype. It can only iterate through iterables (iterables include lists, sets, dict.items(), etc.). 

But ya I would just simplify things. Super hack-y and not the best, but something like below works. Or, better option, loop through the donor list and make sure each donor is in that.
```
def test_write_letters_disk():
    Mailroom.write_letters_disk()
    assert "Retsuko Rarecho" in " ".join(os.listdir())
```